### PR TITLE
Support new Dashboard versioning

### PR DESCRIPTION
**Pull request by breeswish**
### What problem does this PR solve?

Adapt for https://github.com/pingcap-incubator/tidb-dashboard/pull/679

### What is changed and how it works?

Support the new versioning format, as well as support specifying any arbitraty Dashboard commit when building PD.

Note that this PR does not support native Windows (and I think the Windows support should be dropped and prefer WSL instead).

- [x] This PR should be merged only after https://github.com/pingcap-incubator/tidb-dashboard/pull/679 is landed and a release branch is created.

### Check List

Tests

- Manual test

Related changes

- Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by HunDunDM** [comment]
Maybe we can put `describe-dashboard-githash.sh` and `describe-dashboard-internal-version.sh` into one script.

File: scripts/embed-dashboard-ui.sh [hunk 1]
```diff
@@ -1,44 +1,121 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+
+if [ "${DASHBOARD-}" == "0" ] || [ "${DASHBOARD-}" = "SKIP" ]; then
```
[the remark this answers is not in the document]
Comment (breeswish) [reply scripts/embed-dashboard-ui.sh:5]: Yes, good catch

[the remark this answers is not in the document]
Comment (breeswish) [reply scripts/embed-dashboard-ui.sh:5]: I updated Makefile to not introduce a new rule to reduce complexity. Now maybe SKIP is still needed.

**Pull request merged**